src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```diff
@@ -0,0 +1,147 @@
+        this.dialogService = dialogService;
+        this.taskExecutor = taskExecutor;
+        this.preferencesService = preferencesService;
+        this.initialPath = databaseContext.getMetaData().getLaTexFileDirectory(preferencesService.getUser())
+                                          .orElse(preferencesService.getWorkingDir());
+
+        this.root = new SimpleObjectProperty<>();
+        this.checkedFileList = FXCollections.observableArrayList();
+        this.texDirectory = new SimpleStringProperty("");
+        this.noFilesFound = new SimpleBooleanProperty(true);
+        this.searchInProgress = new SimpleBooleanProperty(false);
+        this.successfulSearch = new SimpleBooleanProperty(false);
+
+        texDirectory.set(initialPath.toAbsolutePath().toString());
+    }
+
+    public ObjectProperty<FileNodeViewModel> rootProperty() {
+        return root;
+    }
+
+    private void setRootProperty(FileNodeViewModel fileNode) {
+        root.set(fileNode);
+    }
+
+    public ObservableList<TreeItem<FileNodeViewModel>> getCheckedFileList() {
+        return checkedFileList;
+    }
+
+    public StringProperty texDirectoryProperty() {
+        return texDirectory;
+    }
+
+    public BooleanProperty noFilesFoundProperty() {
+        return noFilesFound;
+    }
+
+    public BooleanProperty searchInProgressProperty() {
+        return searchInProgress;
+    }
+
+    public BooleanProperty successfulSearchProperty() {
+        return successfulSearch;
+    }
+
+    protected void browseButtonClicked() {
+        DirectoryDialogConfiguration directoryDialogConfiguration = new DirectoryDialogConfiguration.Builder()
+                .withInitialDirectory(initialPath).build();
+
+        dialogService.showDirectorySelectionDialog(directoryDialogConfiguration).ifPresent(selectedDirectory -> {
+            texDirectory.set(selectedDirectory.toAbsolutePath().toString());
+            preferencesService.setWorkingDir(selectedDirectory.toAbsolutePath());
+        });
+    }
+
+    protected void searchButtonClicked() {
+        WalkFileTreeTask fileTree = new WalkFileTreeTask(getSearchDirectory());
+
+        BackgroundTask.wrap(fileTree::call)
+                      .onRunning(() -> {
+                          noFilesFound.set(true);
+                          searchInProgress.set(true);
+                          successfulSearch.set(false);
+                      })
+                      .onFinished(() -> searchInProgress.set(false))
+                      .onSuccess(root -> {
+                          setRootProperty(root);
+                          noFilesFound.set(false);
+                          successfulSearch.set(true);
+                      })
+                      .onFailure(dialogService::showErrorDialogAndWait)
+                      .executeWith(taskExecutor);
+    }
+
+    private Path getSearchDirectory() {
+        Path directory = Paths.get(texDirectory.get());
+
+        if (Files.notExists(directory)) {
```

Comment:
I would prefer if you add a proper validation of the file path (that it is a valid and existing directory) and notify the user using a small icon in the text box. Have a look at the other usages of `de.saxsys.mvvmfx.utils.validation` for how this is done.